pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -150,6 +150,35 @@ type Placement struct {
 	// +optional
 	ClusterAffinity *ClusterAffinity `json:"clusterAffinity,omitempty"`
 
+	// OrderedClusterAffinity represents multiple cluster groups.
+	// Each group should have a unique order among all groups within the policy.
+	//
+	// The scheduler will evaluate these groups one by one in strict order, the
+	// group that does not satisfy scheduling restrictions will be ignored that
+	// means all clusters in this group will not be selected unless it also
+	// belongs to the next group(a cluster could belong to multiple groups).
+	//
+	// If none of the groups satisfy the scheduling restrictions, then scheduling
+	// fails, which means no cluster will be selected.
+	//
+	// Note: OrderedClusterAffinity can not co-exist with ClusterAffinity.
+	//
+	// Potential use case 1:
+	// The private clusters in the local data center could be the main group, and
+	// the managed clusters provided by cluster providers could be the secondary
+	// group. So that the Karmada scheduler would prefer to schedule workloads
+	// to the main group and the second group will only be considered in case of
+	// the main group does not satisfy restrictions(like, lack of resources).
```

Comment:
If I have two clusters group, c1(IDC) and c2(public cloud), c1 has 95 replicas, c2 has 1000 replicas.
So with this API, we can schedule 100 replicas to c1(95) and c2(5)?